platforms/software/software-diagnostics/src/main/java/org/gradle/api/plugins/ProjectReportsPlugin.java

```diff
@@ -86,4 +70,16 @@ public void run() {
             projectReportTask.setGroup("reporting");
         });
     }
+
+    @Nonnull
+    private static File getProject(Project project) {
+        return project.getExtensions().getByType(ReportingExtension.class).file("project");
+    }
+
+    private static void mapProjectAndFile(Project project, ConventionReportTask reportTask, String file, String description) {
```

Comment:
What does "map" here mean?

And what does it mean to map a file?

Can we just call this something like `configureReportTask`?

Can we also give the `file` parameter a better name?